File: cpp/04-control-flow/loop_control.cpp
#include <iostream>

int main() {
    // break: stop the loop as soon as a condition is met
    std::cout << "Searching for 4:" << std::endl;
    for (int i = 1; i <= 10; ++i) {
        if (i == 4) {
            std::cout << "Found " << i << ", stopping" << std::endl;
            break;
        }
        std::cout << "Checking " << i << std::endl;
    }

    // continue: skip even numbers, keep going with odd ones
    std::cout << "Odd numbers from 1 to 7: ";
    for (int i = 1; i <= 7; ++i) {
        if (i % 2 == 0) {
            continue;
        }
        std::cout << i << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: lua/04-control-flow/loop_control.lua
-- break stops the loop as soon as a condition is met
print("Finding first multiple of 7:")
for i = 1, 100 do
    if i % 7 == 0 then
        print("Found: " .. i)
        break
    end
end

-- Lua has no 'continue'. Use goto with a label to skip to the next pass.
print("Odd numbers from 1 to 10:")
for i = 1, 10 do
    if i % 2 == 0 then
        goto continue
    end
    io.write(i .. " ")
    ::continue::
end
print()

-- break only exits the INNERMOST loop
print("Products, skipping any that exceed 6:")
for a = 1, 3 do
    for b = 1, 3 do
        if a * b > 6 then
            break
        end
        io.write(a .. "x" .. b .. "=" .. (a * b) .. "  ")
    end
end
print()
